ompi/info/info_memkind.h

```diff
@@ -25,17 +25,26 @@ struct ompi_memkind_t {
 };
 typedef struct ompi_memkind_t ompi_memkind_t;
 
+typedef enum {
+    OMPI_INFO_MEMKIND_ASSERT_UNDEFINED = 0,     // no statement on memkind usage
+    OMPI_INFO_MEMKIND_ASSERT_NO_ACCEL,          // no accelerator memory is used
+    OMPI_INFO_MEMKIND_ASSERT_ACCEL_DEVICE_ONLY, // only accelerator device memory used
+    OMPI_INFO_MEMKIND_ASSERT_ACCEL_ALL          // only accelerator memory (no restrictors) used
```

Comment:
This will include the bits for `OMPI_INFO_MEMKIND_ASSERT_NO_ACCEL`, probably not what we want when we check `flag & OMPI_INFO_MEMKIND_ASSERT_NO_ACCEL`.

Reply:
This is not a bit pattern, but an enum. I was contemplating whether to use a bitpatter instead (in which case your argument would have been correct), but that would have required additional processing afterwards on whether we have only mpi+system enabled or also a gpu one. With an enum we don't have that problem on this layer. Hence, we cannot do `flag & OMPI_INFO_MEMKIND_ASSERT_NO_ACCEL`, but have to do `flag == MPI_INFO_MEMKIND_ASSERT_NO_ACCEL` instead, see e.g. https://github.com/open-mpi/ompi/pull/13095/files#diff-9ab2a505b980955c4d5e74feeb016cb470239023de667666fafe9662f6884498R453

Reply:
Ahh right, I got confused by the assertion bit patterns.